src/pages/Trade/components/ChartCard/Chart.tsx

```diff
@@ -1,50 +1,181 @@
-import React, { FC, useContext } from 'react';
-import { connect } from 'react-redux';
+import React, { FC, useContext, useEffect, useMemo } from 'react';
+import { ConnectedProps, connect } from 'react-redux';
 import styled, { ThemeContext } from 'styled-components';
 import { AreaChart, XAxis, YAxis, Area, Tooltip } from 'recharts';
 import format from 'date-fns/format';
 import { useTranslation } from 'react-i18next';
 import isNumber from 'lodash/isNumber';
+import find from 'lodash/find';
 
 import RechartsResponsiveContainer from 'components/RechartsResponsiveContainer';
 
 import { DataLarge } from 'components/Typography';
 import Spinner from 'components/Spinner';
 
-import { formatCurrencyWithPrecision } from 'utils/formatters';
+import useInterval from 'shared/hooks/useInterval';
+import { formatCurrencyWithSign } from 'utils/formatters';
 
 import { RootState } from 'ducks/types';
-import { SynthPair, SynthDefinitionMap, getAvailableSynthsMap } from 'ducks/synths';
+import { getIsWalletConnected } from 'ducks/wallet/walletDetails';
+import { fetchMyTradesRequest, getMyTrades } from 'ducks/trades/myTrades';
+import { HistoricalTrade } from 'ducks/trades/types';
+import { SynthPair, getAvailableSynthsMap } from 'ducks/synths';
 import { PeriodLabel, PERIOD_IN_HOURS } from 'constants/period';
+import { DEFAULT_REQUEST_REFRESH_INTERVAL } from 'constants/ui';
 
 import { ChartData } from './types';
 
-type StateProps = {
-	synthsMap: SynthDefinitionMap;
+const mapStateToProps = (state: RootState) => ({
+	synthsMap: getAvailableSynthsMap(state),
+	trades: getMyTrades(state),
+	isWalletConnected: getIsWalletConnected(state),
+});
+
+const mapDispatchToProps = {
+	fetchMyTradesRequest,
 };
 
-type Props = {
+const connector = connect(mapStateToProps, mapDispatchToProps);
+
+type PropsFromRedux = ConnectedProps<typeof connector>;
+
+type ChartProps = PropsFromRedux & {
 	data: ChartData;
 	isLoading: boolean;
 	period: PeriodLabel;
 	synthPair: SynthPair;
 };
-type ChartProps = StateProps & Props;
 
-const Chart: FC<ChartProps> = ({ synthPair: { quote }, data, isLoading, period, synthsMap }) => {
+const Chart: FC<ChartProps> = ({
+	isWalletConnected,
+	fetchMyTradesRequest,
+	trades,
+	synthPair: { quote },
+	data,
+	isLoading,
+	period,
+	synthsMap,
+}) => {
 	const { colors } = useContext(ThemeContext);
 	const { t } = useTranslation();
 	const synthSign = synthsMap[quote.name] && synthsMap[quote.name].sign;
 
+	useEffect(() => {
+		if (isWalletConnected) {
+			fetchMyTradesRequest();
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [isWalletConnected]);
```

Comment:
you could put `fetchMyTradesRequest` and then you don't need to disabled rule

Reply:
I thought it was already there, but for sure I can change that for this one